File: app/src/main/java/life/plank/juna/zone/data/network/model/ScrubberData.java
package life.plank.juna.zone.data.network.model;

import lombok.Data;

@Data
public class ScrubberData {
    private int millisecondsX;
    private int interactionY;
    private MatchEvent event;

    public ScrubberData(int millisecondsX, int interactionY, String eventType, boolean isHomeTeam) {
        this.millisecondsX = millisecondsX;
        this.interactionY = interactionY;
        this.event = new MatchEvent(eventType, isHomeTeam);
    }
}

File: app/src/main/java/life/plank/juna/zone/data/model/ScrubberData.kt
package life.plank.juna.zone.data.model

import android.os.Parcelable
import kotlinx.android.parcel.IgnoredOnParcel
import kotlinx.android.parcel.Parcelize

@Parcelize
data class ScrubberData(
        var millisecondsX: Long,
        var interactionY: Long,
        var eventType: String,
        var isHomeTeam: Boolean
) : Parcelable {
    @IgnoredOnParcel
    var event: MatchEvent = MatchEvent(this.isHomeTeam, this.eventType)

    constructor(millisecondsX: Long, interactionY: Long, event: MatchEvent) : this(millisecondsX, interactionY, event.eventType, event.isHomeTeam) {
        this.event = event
    }
}